skyportal/handlers/api/observation_plan.py

```diff
@@ -84,6 +85,136 @@
 log = make_log('api/observation_plan')
 
 
+def post_survey_efficiency_analysis(
+    survey_efficiency_analysis, plan_id, user_id, session
+):
+    """Post survey efficiency analysis to database.
+
+    Parameters
+    ----------
+    survey_efficiency_analysis: dict
+        Dictionary describing survey efficiency analysis
+    plan_id: int
+        SkyPortal ID of Observation plan request
+    user_id : int
+        SkyPortal ID of User posting the GcnEvent
+    session: sqlalchemy.Session
+        Database session for this transaction
+    """
+
+    status_complete = False
+    while not status_complete:
+        observation_plan_request = (
+            session.query(ObservationPlanRequest)
+            .options(
+                joinedload(ObservationPlanRequest.observation_plans)
+                .joinedload(EventObservationPlan.planned_observations)
+                .joinedload(PlannedObservation.field)
+            )
+            .get(plan_id)
+        )
+        status_complete = observation_plan_request.status == "complete"
+
+        if not status_complete:
+            time.sleep(30)
+
+    allocation = session.query(Allocation).get(observation_plan_request.allocation_id)
+    localization = session.query(Localization).get(
+        observation_plan_request.localization_id
+    )
```

Comment:
Can we use `select` here instead of `query`?

Reply:
yes